sklearn/metrics/_dist_metrics.pyx.tp

```diff
@@ -42,7 +42,13 @@ cdef extern from "arrayobject.h":
                                      int typenum, void* data)
 
 
-from libc.math cimport fabs, sqrt, exp, pow, cos, sin, asin
+# ddot from CBLAS (cf sklearn/linear_model/cd_fast.pyx:102-117)
+cdef extern from "cblas.h":
+    double ddot "cblas_ddot"(int N, double *X, int incX, double *Y, int incY
+                             ) nogil
```

Comment:
Internally, we define and uses interfaces for BLAS' which handle various precision using  Cython fused-types. For instance, we have a `_dot` interface for the `sdot` and `ddot` BLAS' subroutines:

https://github.com/scikit-learn/scikit-learn/blob/b571d64066dd13b6f9b5eae399c6a20db39e5a83/sklearn/utils/_cython_blas.pyx#L20-L26

`_dot` must be imported and used inplace of `ddot`.

The import, that must be placed at l.48 of the original file, is:  
```
from ..utils._cython_blas cimport _gemm
```

Reply:
In the above line, I replaced `_gemm` by `_dot`:

```cython
from ..utils._cython_blas cimport _dot
```

and then inserted that at line 50, after removing the former `cdef extern` on line 46-48.

Reply:
Good catch!